### ISSUE TITLE
Audi account gets disabled

### ISSUE THREAD

**Issue by tomasenskede** [audiconnect/audi_connect_ha#292]
I have experienced that I lost control of my car now and then. Audi Connect investigated and identified the root cause: excessive logins to my account via the API. When a certain threshold is reached, the account gets disabled. I've inquired about the permissible number of logins per day or week. 

Query: Is there a way to adjust the update interval configuration?

**Comment by cdnninja** [comment]
How recent was this?

**Comment by fhloston** [comment]
Same happened couple of days ago to my account. Audiconnect app also cannot refresh data.

How is the re-enablement procedure of the account? @tomasenskede

**Comment by tomasenskede** [comment]
Don't know yet, I have asked Audi about this, I will share the answer asap

**Comment by cpohli** [comment]
> Query: Is there a way to adjust the update interval configuration?

If you delete the car in the integration (just the one car, not the hole integration) and add it back again, you can specify the refreshment interval. It's set to 10 minutes by defaukt, you can crank this up ... It's the last option in the email-password-request window when you add a new car.

**Comment by oleg-d** [comment]
I think the default 10 mins could be too often. I have mine set to 60 mins currently, but even 30 mins was probably ok if I recall I had tried that before.

**Comment by fhloston** [comment]
I had it running with 1min for more than 2 years with my previous e-tron 55. 

Now with the Q8 e-tron I noticed the account getting blocked when the shorttrip and longtrip sensors were queried and after the uncommenting of those it worked again.

I was using the SoC sensor with a REST automation to send the SoC to the ABRP api, which has worked for over 2 years.

I hope they are only now locking down due to api errors. I would really not like to only be able to get updated data every 30mins or so.

**Comment by tomasenskede** [comment]
After deleting and re-adding my account, I was prompted to set an interval. I chose 60 minutes, but I only received updates for the door lock and model, not the SOC. Any thoughts on why this might be happening?

**Comment by SamKirsch10** [comment]
Are you in the US? I noticed this as well for me. I had to change the endpoints as they were using the EMEA ones. See my PR if you want to try it out.

**Comment by tomasenskede** [comment]
> After deleting and re-adding my account, I was prompted to set an interval. I chose 60 minutes, but I only received updates for the door lock and model, not the SOC. Any thoughts on why this might be happening?

After the reboot, numerous sensors became available. I'm situated in Sweden, Europe. Let's test if I can maintain the account's availability over time by updating my information only once every 60 minutes. Fingers crossed!

**Comment by fhloston** [comment]
My account came back, I can refresh in the app again. I just disabled the audiconnect integration for a couple of days.

**Comment by tomasenskede** [comment]
Mine came back for a while and now its blocket again. I have installed the audiconnect integration with 60 min refresh intervals... but it still to often. I got this answer from Audi;

_We have escalated the matter to our technicians again, and unfortunately, there is nothing more we can do from our end. They were also unable to provide further information regarding your question about how many times ASI can connect to the SOS system.

We will forward your message to our technicians again.

When we analyzed your Audi Connect case, we found that your myAudi login credentials were used to access our servers an unusually large number of times. One reason for this could be the use of a third-party app (e.g., for charging planning for electric vehicles) or similar, which has been allowed to use your personal car login credentials. When such a situation is detected, the car's Audi Connect services will automatically be blocked for the rest of the day. This also means that it will no longer be possible for you to use the official myAudi app for vehicle functions on that day.

As a precaution to avoid this type of problem in the future, we kindly ask you to change your myAudi password to prevent unauthorized third-party access. Alternatively, you can revoke your previous consent for third parties to use your personal login credentials and only use the official myAudi app for your connectivity services. Unfortunately, we cannot do more than this.

We appreciate your understanding._

... so I will change my password and see if its resolves my issue...

**Comment by fhloston** [comment]
I assume that the access is blocked faster when your api queries run into errors.

Interestingly I have not yet read about similar errors within the evcc issue tracker. They also poll the audi api to retrieve at least SoC.

**Comment by tomasenskede** [comment]
Is their a log where I can see the connection attempts that the connection makes?

**Comment by devslash0** [comment]
I also get blocked out from time to time now. Did not happend before they upgraded API I think. I get Too Many Requests and login to audi service failed. I use default 10 timeout. I actually would like 1 min timeout but see that that can be a problem :P

**Comment by tomasenskede** [comment]
Then we are experiencing the same behaviour... Where do you find these settings? Would be good to find a working config...

**Comment by fhloston** [comment]
I think it is their way of trying to prevent third parties from accessing their api.

**Comment by oleg-d** [comment]
I am on 60 mins refresh interval and now got errors too the other day, something which has never happened before with such long interval. Had to disable the integration in HA for a day.

> I think it is their way of trying to prevent third parties from accessing their api.

Time to buy another car.

**Comment by ngluvakov** [comment]
It's time to start thinking about the OBD update...

**Comment by tomasenskede** [comment]
after 60 mins intervall and new password to audi api it has been stable for a week soon... fingers crossed...

**Comment by coreywillwhat** [comment]
#325 Manual service call to request cloud update
#321 Refresh Interval Options from integration > configure screen

**Comment by coreywillwhat** [comment]
#326 Turning Active Polling on or off

**Comment by coreywillwhat** [comment]
closing this issue as you can now turn active polling on/off and change polling interval. You can also update from the cloud whenever using the `refresh_cloud_data` service. 

If you continue to experience issues, please open a new issue.
